packages/react-dom/src/__tests__/ReactServerRenderingHydration-test.js

```diff
@@ -531,4 +533,31 @@ describe('ReactDOMServerHydration', () => {
       'Warning: Did not expect server HTML to contain a <p> in <div>',
     );
   });
+
+  // @gate enableCustomElementPropertySupport
+  it('should warn when hydrating read-only properties', () => {
+    const readOnlyProperties = [
+      'offsetParent',
+      'offsetTop',
+      'offsetLeft',
+      'offsetWidth',
+      'offsetHeight',
+      'isContentEditable',
+    ];
+    readOnlyProperties.forEach(readOnlyProperty => {
+      const props = {};
+      props[readOnlyProperty] = 'hello';
+      const jsx = React.createElement('my-custom-element', props);
+      const element = document.createElement('div');
+      element.innerHTML = ReactDOMServer.renderToString(jsx);
+      expect(() => ReactDOM.hydrate(jsx, element)).toErrorDev(
+        `Warning: Assignment to read-only property will result in a no-op: \`${readOnlyProperty}\``,
+      );
+
+      // Ensure the test fails to satisfy @gate when the flag is turned off and we aren't in development.
```

Comment:
This seems confusing. I suggest instead to always run the test (if the prod behavior is the same) but to keep the `toErrorDev` in inline `if (gate(...))` checks. So `if (gate(...)) { expect(() => hydrate()).toErrorDev(...) } else { hydrate() }`

Reply:
That did the trick, thanks!